### PR TITLE
language: nicer error for package that depends on itslef.

### PR DESCRIPTION
**Pull request by robin-da**
We add a consistency check for package related config fields in the
daml.yaml to avoid confusing errors when building the package.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/damlc/lib/DA/Cli/Damlc.hs [hunk 1]
```diff
@@ -390,7 +390,18 @@ parseProjectConfig project = do
...
     Right $ PackageConfigFields name main exposedModules version dependencies sdkVersion
+  where
+    consistencyCheck n ver deps
+        | n <> "-" <> ver `elem` (map takeBaseName deps) =
+            Left $
+            ConfigFieldInvalid
+                "name"
+                ["name"]
+                ("Package " <> n <>
+                 " can not depend on itself. Either rename the package or remove it from the dependencies field.")
```
Comment (cocreature) [inline compiler/damlc/lib/DA/Cli/Damlc.hs]:
```suggestion
                 " can not depend on a package with the same name. Either rename the package or remove it from the dependencies field.")
```

File: compiler/damlc/lib/DA/Cli/Damlc.hs [hunk 2]
```diff
@@ -390,7 +390,18 @@ parseProjectConfig project = do
     dependencies <-
         queryProjectConfigRequired ["dependencies"] project
     sdkVersion <- queryProjectConfigRequired ["sdk-version"] project
+    consistencyCheck name version dependencies
     Right $ PackageConfigFields name main exposedModules version dependencies sdkVersion
+  where
+    consistencyCheck n ver deps
+        | n <> "-" <> ver `elem` map takeBaseName deps =
```
Comment (cocreature) [inline compiler/damlc/lib/DA/Cli/Damlc.hs]: I don’t think this check is sufficient. You are relying on people using the default DAR name which isn’t a valid assumption (e.g. that wouldn’t have caught my error). I think we should ask GHC for the name of the packages in the package db.

Comment (robin-da) [reply compiler/damlc/lib/DA/Cli/Damlc.hs]: I'll close this PR and solve it with a lookup in the package db.

Comment (cocreature) [reply compiler/damlc/lib/DA/Cli/Damlc.hs]: Thanks!

**Pull request closed without merging**